sklearn/utils/tests/test_extmath.py

```diff
@@ -161,6 +162,88 @@ def test_randomized_svd_low_rank_all_dtypes(dtype):
     check_randomized_svd_low_rank(dtype)
 
 
+@pytest.mark.parametrize('dtype',
+                         (np.int32, np.int64, np.float32, np.float64))
+def test_randomized_eigsh(dtype):
+    """Test that `_randomized_eigsh` returns the appropriate components"""
+
+    rng = np.random.RandomState(42)
+    X = np.diag(np.array([1., -2., 0., 3.], dtype=dtype))
+    # random rotation that preserves (but invert signs) the eigenvalues of X:
+    X = X @ np.linalg.qr(rng.normal(size=X.shape))[0]
```

Comment:
Actually if you want to get a real rotation SO(4) instead of just a random orthonormal matrix you can directly use:

```python
from scipy.stats import special_ortho_group

X = X @ special_ortho_group.rvs(X.shape[0], random_state=0)
```

Reply:
Ah thanks this makes more sense ! I was confused on why the rotation would flip signs.

Reply:
Oh we made a mistake, `R * X * R^T` is the formula to use

Reply:
So I switched back to using `qr` since it works already fine